knn/knn.go

```diff
@@ -7,7 +7,7 @@ import (
 	"errors"
 	"fmt"
 
-	"github.com/gonum/matrix"
+	matrix "gonum.org/v1/gonum/mat"
```

Comment:
duplicate declaration. How about removing this import and fixing the usage of it.